framework/Samplers/CustomSampler.py

```diff
@@ -100,6 +100,10 @@ def _readMoreXMLbase(self,xmlNode):
           self.toBeSampled[name] = 'custom'
         else:
           self.dependentSample[name] = funct.text.strip()
+      elif child.tag == 'constant':
+        name = child.attrib['name']
+        value = float(child.text)
```

Comment:
why you convert it here? Should you use the InputData to check it?

Reply:
The `CustomSampler` does not yet use the `InputData` system, as far as I can tell. There is a comment on line 87 noting this, which will be great to fix someday.

Reply:
I updated the `readMoreXMLbase` for the `CustomSampler` to use `InputData`, with the exception of the `readSamplerInit` which needs to be fixed across many samplers.